composer/trainer/devices/utils.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Device-related helper methods and utilities."""
```

Comment:
I dont think we want an additional 'utils" file here -- otherwise LGTM

Reply:
Big +1 please no more utils files :(

Reply:
Good to know. Should I move these functions into `utils.dist.py`?

Reply:
I'm pro adding a `device.py` file in `composer/utils`, and then making sure its setup so we can do `from composer.utils import get_device`. @hanlint thoughts? 

Dist is usually imported for distributed stuff, and most calls are `dist.fn_name`. Since this is imported directly, I'd rather not do it there. The other option is put in `misc.py`

Reply:
Maybe we keep this PR scoped to not movig the helper functions around, and just keep them in the trainer for now? We can clean the rest up when you work on the Python API

Reply:
I'm also fine not moving it

Reply:
Been banging my head against this circular import issue. Would love to know if anyone has any elegant suggestions:
```
Traceback (most recent call last):
  File "/Users/mleavitt/Repos/mosaic/venvs/composer/lib/python3.8/site-packages/sphinx/config.py", line 340, in eval_config_file
    exec(code, namespace)
  File "/Users/mleavitt/Repos/mosaic/composer/docs/source/conf.py", line 42, in <module>
    import composer
  File "/Users/mleavitt/Repos/mosaic/composer/composer/__init__.py", line 7, in <module>
    from composer.core import Algorithm, Callback, DataSpec, Engine, Evaluator, Event, State, Time, Timestamp, TimeUnit
  File "/Users/mleavitt/Repos/mosaic/composer/composer/core/__init__.py", line 13, in <module>
    from composer.core.data_spec import DataSpec, ensure_data_spec
  File "/Users/mleavitt/Repos/mosaic/composer/composer/core/data_spec.py", line 14, in <module>
    from composer.utils.iter_helpers import ensure_tuple
  File "/Users/mleavitt/Repos/mosaic/composer/composer/utils/__init__.py", line 8, in <module>
    from composer.utils.checkpoint import load_checkpoint, save_checkpoint
  File "/Users/mleavitt/Repos/mosaic/composer/composer/utils/checkpoint.py", line 21, in <module>
    from composer.utils import dist, reproducibility
  File "/Users/mleavitt/Repos/mosaic/composer/composer/utils/dist.py", line 45, in <module>
    from composer.device import get_device
  File "/Users/mleavitt/Repos/mosaic/composer/composer/device.py", line 10, in <module>
    from composer.trainer.devices import Device, DeviceCPU, DeviceGPU, DeviceMPS, DeviceTPU
  File "/Users/mleavitt/Repos/mosaic/composer/composer/trainer/__init__.py", line 7, in <module>
    from composer.trainer.trainer import Trainer
  File "/Users/mleavitt/Repos/mosaic/composer/composer/trainer/trainer.py", line 30, in <module>
    from composer.algorithms import GradientClipping
  File "/Users/mleavitt/Repos/mosaic/composer/composer/algorithms/__init__.py", line 40, in <module>
    from composer.algorithms.alibi import Alibi
  File "/Users/mleavitt/Repos/mosaic/composer/composer/algorithms/alibi/__init__.py", line 11, in <module>
    from composer.algorithms.alibi.alibi import Alibi, apply_alibi
  File "/Users/mleavitt/Repos/mosaic/composer/composer/algorithms/alibi/alibi.py", line 14, in <module>
    from composer.core import Algorithm, Event, State
  File "/Users/mleavitt/Repos/mosaic/composer/composer/core/State.py", line 24, in <module>
    from composer.utils import batch_get, batch_set, dist, ensure_tuple, is_model_deepspeed
ImportError: cannot import name 'ensure_tuple' from partially initialized module 'composer.utils' (most likely due to a circular import) (/Users/mleavitt/Repos/mosaic/composer/composer/utils/__init__.py)
```

Reply:
hm... this is annoying.... You can 1) put import `Device` around `if TYPE_CHECKING` and 2) move the other device imports inside the function for now

Reply:
tyvm, fixed!